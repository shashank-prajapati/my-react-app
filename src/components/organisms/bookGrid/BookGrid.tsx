import React from "react";
import { Box, Grid } from "@mui/material";
import BookCard from "../../molecules/bookCard/BookCard";

interface bookData {
  id: number;
  coverImage: string;
  bookTitle: string;
  bookAuthor: string;
  time: string;
  isFinished: boolean;
}

type Props = {
  booksData: Array<bookData>;
  cardButtonClick: any;
};

function BookGrid(props: Props) {
  const { booksData, cardButtonClick } = props;

  const booksGridItems = booksData.map((book) => {
    let buttonText = "Finish Reading";
    if (book.isFinished) {
      buttonText = "Read Again";
    }
    return (
      <Grid item xs={4} display='flex' justifyContent='center'>
        <BookCard
          key={book.id}
          coverImage={book.coverImage}
          bookTitle={book.bookTitle}
          bookAuthor={book.bookAuthor}
          time={book.time}
          isButtoned={true}
          btnText={buttonText}
          handleClick={cardButtonClick}
          index={book.id}
        />
      </Grid>
    );
  });

  return (
    <div>
      <Grid container columnSpacing={10} rowGap={4}>
        {booksGridItems}
      </Grid>
    </div>
  );
}

export default BookGrid;
