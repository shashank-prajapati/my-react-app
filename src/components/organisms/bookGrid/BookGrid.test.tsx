import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import BookGrid from "./BookGrid";

import BBCoverImage from "../../../media/cover_image/Boss.png";
import ETECoverImage from "../../../media/cover_image/ETE.png";
import DHTACoverImage from "../../../media/cover_image/DHTA.png";
import FOFCoverImage from "../../../media/cover_image/FOF.png";
import LOSCoverImage from "../../../media/cover_image/LOS.png";
import LYBCoverImage from "../../../media/cover_image/LYB.png";
import TLCCoverImage from "../../../media/cover_image/TLC.png";
import EBFBCoverImage from "../../../media/cover_image/EBFB.png";
import DropshippingCoverImage from "../../../media/cover_image/DropShipping.png";
import EntrepreneurshipCI from "../../../media/cover_image/Entrepreneurship.png";

const booksData = [
    {
      id: 1,
      coverImage: BBCoverImage,
      bookTitle: "Being Boss",
      bookAuthor: "Kathleen Shannon and Emily...",
      time: "13",
      isFinished:false
    },
    {
      id: 2,
      coverImage: ETECoverImage,
      bookTitle: "Employee to Entrepreneur",
      bookAuthor: "Steve Glaveski",
      time: "15",
      isFinished:false
    },
    {
      id: 3,
      coverImage: DHTACoverImage,
      bookTitle: `Doesnâ€™t Hurt to Ask`,
      bookAuthor: "Trey Gowdy",
      time: "13",
      isFinished:false
    },
    {
      id: 4,
      coverImage: FOFCoverImage,
      bookTitle: `The Fate of Food`,
      bookAuthor: "Amanda Little",
      time: "12",
      isFinished:false
    },
    {
      id: 5,
      coverImage: LOSCoverImage,
      bookTitle: `Lives of the Stoics`,
      bookAuthor: "Ryan Holiday, Stephen Hansel",
      time: "12",
      isFinished:false
    },
    {
      id: 6,
      coverImage: LYBCoverImage,
      bookTitle: `Loving Your Business`,
      bookAuthor: "Debbie King",
      time: "12",
      isFinished:false
    },
    {
      id: 7,
      coverImage: TLCCoverImage,
      bookTitle: `The Lonely Century`,
      bookAuthor: "Noreena Hertz",
      time: "15",
      isFinished:false
    },
    {
      id: 8,
      coverImage: EBFBCoverImage,
      bookTitle: `Eat Better, Feel Better`,
      bookAuthor: "Giada De Laurentiis",
      time: "13",
      isFinished:false
    },
    {
      id: 9,
      coverImage: DropshippingCoverImage,
      bookTitle: `Dropshipping`,
      bookAuthor: "James Moore",
      time: "12",
      isFinished:false
    },
    {
      id: 10,
      coverImage: EntrepreneurshipCI,
      bookTitle: "Beyond Entrepreneurship",
      bookAuthor: "Jim Collins & Bill Lazier",
      time: "13", 
      isFinished:true
    }
  ];
  

describe("Testing Book Grid", () => {
  it("should match the Book Grid snapshot", () => {
    const { container } = render(<BookGrid booksData={booksData} cardButtonClick={undefined} />);
    expect(container).toMatchSnapshot();
  });
  
  it("should call on click method of finished reading", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <BookGrid
        booksData={booksData}
        cardButtonClick={handleClick}
      ></BookGrid>
    );
    fireEvent.click(screen.getAllByText("Finish Reading")[0]);
    expect(handleClick).toHaveBeenCalled();
  });

  it("should call on click method of read again", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <BookGrid
        booksData={booksData}
        cardButtonClick={handleClick}
      ></BookGrid>
    );
    fireEvent.click(screen.getAllByText("Read Again")[0]);
    expect(handleClick).toHaveBeenCalled();
  });
});
