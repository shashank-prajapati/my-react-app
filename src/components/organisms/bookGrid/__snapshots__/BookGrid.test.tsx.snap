// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Book Grid should match the Book Grid snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container css-1auibvu-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/Boss.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Being Boss
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Kathleen Shannon and Emily...
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        13
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/ETE.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Employee to Entrepreneur
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Steve Glaveski
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        15
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/DHTA.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Doesnâ€™t Hurt to Ask
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Trey Gowdy
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        13
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/FOF.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    The Fate of Food
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Amanda Little
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        12
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/LOS.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Lives of the Stoics
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Ryan Holiday, Stephen Hansel
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        12
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/LYB.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Loving Your Business
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Debbie King
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        12
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/TLC.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    The Lonely Century
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Noreena Hertz
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        15
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/EBFB.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Eat Better, Feel Better
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Giada De Laurentiis
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        13
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/DropShipping.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Dropshipping
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    James Moore
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        12
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Finish Reading
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1fttiv4-MuiGrid-root"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
            data-testid="Card"
          >
            <img
              alt="Book"
              height="100%"
              src="/assets/cover_image/Entrepreneurship.png"
              style="display: block;"
              width="100%"
            />
            <div>
              <div
                class="MuiBox-root css-9r861l"
              >
                <div
                  class="MuiBox-root css-3jmp7d"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
                  >
                    Beyond Entrepreneurship
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
                  >
                    Jim Collins & Bill Lazier
                  </p>
                  <div>
                    <div
                      class="MuiBox-root css-5as2nd"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                        focusable="false"
                      >
                        clockIcon.svg
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                      >
                        13
                        -minute read
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-eb3voy"
                  />
                </div>
                <div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Read Again
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
